util/archive_builder.go

```diff
@@ -316,11 +317,11 @@ func (builder MtaArchiveBuilder) getModulesPaths(deploymentDescriptorResources [
 	return result, nil
 }
 
-func normalizePaths(elementsPaths map[string]string) map[string]string {
+func normalizePathsForUnix(elementsPaths map[string]string) map[string]string {
 	normalizedPaths := make(map[string]string, len(elementsPaths))
 	for key, value := range elementsPaths {
 		if value != "" {
-			normalizedPaths[key] = filepath.Join(key, filepath.Base(value))
+			normalizedPaths[key] = key + unixPathSeparator + filepath.Base(value)
```

Comment:
I would use some kind of formatter here. For instance, zipEntryNormalizedPathTemplate = "%s/%s" and with fmt.Sprintf to apply it here. This way the ugly + signs will be removed.
Also, remove the unixPathSeparator when you implement it that way